raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -682,41 +674,35 @@ def _on_ice_gathering_state_change(
         for candidate in rtc_ice_candidates:
             candidate = {
                 "candidate": f"candidate:{candidate_to_sdp(candidate)}",
-                "sdpMid": candidate.sdpMid if candidate.sdpMid is not None else SDP_MID_DEFAULT,
+                "sdpMid": candidate.sdpMid if candidate.sdpMid is not None else _SDP_MID_DEFAULT,
                 "sdpMLineIndex": candidate.sdpMLineIndex is not None
                 if candidate.sdpMLineIndex
-                else SDP_MLINE_INDEX_DEFAULT,
+                else _SDP_MLINE_INDEX_DEFAULT,
             }
             candidates.append(candidate)
 
-        wrap_callback(
-            callback=candidates_callback,
-            candidates=candidates,
-            partner_address=rtc_partner.partner_address,
-        )
+        wrap_callback(callback=candidates_callback, candidates=candidates)
 
 
-def _on_ice_connection_state_change(
-    rtc_partner: _RTCPartner, closed_callback: Callable[[Address], None]
-) -> None:
-    ice_connection_state = rtc_partner.peer_connection.iceConnectionState
-    rtc_partner.log.debug("Ice connection state changed", signaling_state=ice_connection_state)
+def _on_ice_connection_state_change(conn: _RTCConnection) -> None:
+    ice_connection_state = conn.peer_connection.iceConnectionState
+    conn.log.debug("ICE connection state changed", signaling_state=ice_connection_state)
 
-    if ice_connection_state in [ICEConnectionState.CLOSED.value, ICEConnectionState.FAILED.value]:
-        asyncio.create_task(rtc_partner.reset())
-        wrap_callback(callback=closed_callback, partner_address=rtc_partner.partner_address)
+    if ice_connection_state in [
+        _ICEConnectionState.CLOSED.value,
+        _ICEConnectionState.FAILED.value,
+    ]:
+        asyncio.create_task(conn.reset())
+        asyncio.create_task(conn.ice_connection_closed())
```

Comment:
Why does this need to be its own coroutine? looking at the underlying code it does not have any `await` statement and it only schedules a greenlet which could be done synchronously.